tests/DataFactories/Newsletter.php

```diff
@@ -87,6 +109,19 @@ public function withDeleted() {
     return $this;
   }
 
+  /**
+   * @param \MailPoet\Models\Segment[] $segments
+   * @return Newsletter
+   */
+  public function withSegments(array $segments) {
+    $ids = [];
```

Comment:
I would use `$this->segments` directly here instead of introducing a new variable.